### PR TITLE
nvuApp: admin should be able to see all questions

### PR DESCRIPTION
**Pull request by qnislavnachev**
there is a hardcore implemented admin user that is able to see
all questions uploaded by all users, the information for questions is
in the code implemeted by InMemoryRepository. In this PR will be added persistant
repository.

Related to #5

### CODE REVIEW

**Comment by qnislavnachev** [comment]
PR need a review

File: src/test/java/com/clouway/nvuapp/core/AdminQuestionListHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        System.out.println(page);
+        assertThat(page, containsString("User-1234 How are you today?"));
+        assertThat(page, containsString("User-1234 How you feel today?"));
+
+        assertThat(page, containsString("User-0987 How are you today?"));
+        assertThat(page, containsString("User-0987 How you feel today?"));
+    }
+
+    @Test
+    public void NoQuestionsInRepository() throws Exception {
```
Comment (mgenov) [inline src/test/java/com/clouway/nvuapp/core/AdminQuestionListHandlerTest.java]: methods in java should start in lower case character.

File: src/test/java/com/clouway/nvuapp/core/AdminQuestionListHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                        new Question("1234", "CAT2", 23, 1, 1, 1, "User-1234 How you feel today?", "I feel Good", "I feel bad", "I feed unusual")
+                ),
+                "0987", Lists.newArrayList(
+                        new Question("0987", "CAT1", 23, 1, 1, 1, "User-0987 How are you today?", "I feel Good", "I feel bad", "I feed unusual"),
+                        new Question("0987", "CAT2", 23, 1, 1, 1, "User-0987 How you feel today?", "I feel Good", "I feel bad", "I feed unusual")
+                )
+        ));
+        AdminQuestionListHandler questionListHandler = new AdminQuestionListHandler("admin", repository);
+        Response response = questionListHandler.handle(request);
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
```
Comment (mgenov) [inline src/test/java/com/clouway/nvuapp/core/AdminQuestionListHandlerTest.java]: You are having multiple duplications of ByteArrayOutputStream which should be avoid. 

In fserve library it's used as 

```java
assertThat(new RsPrint(response).print(), containsString("User-1234 How are you today?")); 
```

but you can consider and better options.

File: src/test/java/com/clouway/nvuapp/core/AdminQuestionListHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        Response response = questionListHandler.handle(request);
+        ByteStreams.copy(response.body(), out);
+
+        String page = out.toString();
+
+        assertThat(page, containsString("Няма добавени въпроси до момента"));
+    }
+
+    @Test
+    public void tutorIdIsNotAdmin() throws Exception {
```
Comment (mgenov) [inline src/test/java/com/clouway/nvuapp/core/AdminQuestionListHandlerTest.java]: tutorIsNotAnAdmin

**Comment by qnislavnachev** [comment]
made some fixes to test

File: src/test/java/com/clouway/nvuapp/core/AdminQuestionListHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        QuestionRepository repository = new InMemoryQuestionRepository(Collections.<String, List<Question>>emptyMap());
+        AdminQuestionListHandler questionListHandler = new AdminQuestionListHandler("notAdmin", repository);
+
+        Response response = questionListHandler.handle(request);
+        String uri = response.headers().get("Location");
+
+        assertThat(uri, equalTo("/home"));
+    }
+
+    private String toString(Response response) {
```
Comment (mgenov) [inline src/test/java/com/clouway/nvuapp/core/AdminQuestionListHandlerTest.java]: This method removes duplication only in methods of this test but not in other tests.

**Comment by qnislavnachev** [comment]
created a class that is used to avoid byteArrayStreams in test

**Pull request merged**